### PR TITLE
Hunk staging and LICENSE update

### PR DESCRIPTION
**Pull request by isacikgoz**
- introduces hunk staging for #27 
- changes license from MIT to BSD-3-Clause
- diff of staged files can be observed

### CODE REVIEW

**Comment by ybbond** [comment]
I think this PR satisfies my issue (Fixes #37 ) right?

you just need to add the:
>If you are using macOS, actually, you don't need to build libgit2 v27 because you can install it via`brew install libgit2` then you can just download project with `go get github.com/isacikgoz/gitin` and start hacking.

File: README.md [hunk 1]
```diff
@@ -40,27 +40,35 @@ Commands:
...
-  6. run the script `./script/install-libgit2.sh`
+- Requires gitlib2 v27 and `git2go`. See the project homepages for more information about build instructions. For gitin you can simply;
+  - macOS:
+    1. install libgit2 via `brew install libgit2` (consider that libgit2.v27 is required)
+  - Linux and macOS(if you want to build your own):
+    1. download git2go; `go get -d gopkg.in/libgit2/git2go.v27`
+    2. make sure you have `cmake`, `pkg-config` and `libssl-dev` installed
+    3. `cd` into `$GOPATH/src/gopkg.in/libgit2/git2go.v27`
+    4. initialize submodules by running `git submodule update --init`
+    5. change the libigt2 version to your version (in this case its 0.27) in the install script (`script/install-libgit2.sh`)
```
Comment (ybbond) [inline README.md]: I think this instruction is not clear enough, maybe: 
```suggestion
    5. change the libigt2 version to your version (in this case its 0.27) in the install script (e.g. `nano script/install-libgit2.sh` or `vim script/install-libgit2.sh`) and change `LG2VER` to 0.27.0
```

Comment (isacikgoz) [reply README.md]: thank you @ybbond  this is more clear

**Comment by isacikgoz** [comment]
fixes #27 
fixes #36 
fixes #37

**Pull request merged**